### PR TITLE
Storybook version update

### PR DESCRIPTION
**Pull request by EugeneTseitlin**
Issue Number: N/A

## PR Details

Updated dependencies to run storybook

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-design-system/package.json [hunk 1]
```diff
@@ -55,17 +57,20 @@
     "eslint-plugin-react-hooks": "4.6.0",
     "formik": "2.2.9",
     "prettier": "2.1.2",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
```
Comment (tupe12334) [inline packages/amplication-design-system/package.json:61]: Why do this packages moved from peer

File: packages/amplication-design-system/package.json [hunk 2]
```diff
@@ -31,22 +31,24 @@
...
+    "@storybook/react": "^6.5.10",
+    "@storybook/testing-library": "^0.0.13",
     "@types/classnames": "2.3.1",
     "@types/lodash": "4.14.178",
     "@types/react": "17.0.34",
     "@typescript-eslint/eslint-plugin": "5.9.1",
     "@typescript-eslint/parser": "5.9.1",
     "babel-loader": "8.2.5",
-    "css-loader": "6.7.1",
+    "css-loader": "^6.7.1",
```
Comment (tupe12334) [inline packages/amplication-design-system/package.json:51]: Why is this not anchor?

File: packages/amplication-design-system/package.json [hunk 3]
```diff
@@ -55,17 +57,20 @@
...
     "rollup-plugin-typescript2": "0.32.1",
     "sass": "1.35.2",
-    "sass-loader": "12.6.0",
-    "style-loader": "3.3.1",
+    "sass-loader": "^12.6.0",
+    "style-loader": "^3.3.1",
+    "styled-components": "^5.3.3",
     "typescript": "3.9.10",
-    "webpack": "5.65.0"
+    "webpack": "^5.74.0"
```
Comment (tupe12334) [inline packages/amplication-design-system/package.json:73]: Why is this not anchor?

File: packages/amplication-design-system/package.json [hunk 4]
```diff
@@ -55,17 +57,20 @@
...
     "rollup": "2.63.0",
     "rollup-plugin-copy": "3.4.0",
     "rollup-plugin-postcss": "3.1.8",
     "rollup-plugin-typescript2": "0.32.1",
     "sass": "1.35.2",
-    "sass-loader": "12.6.0",
-    "style-loader": "3.3.1",
+    "sass-loader": "^12.6.0",
+    "style-loader": "^3.3.1",
+    "styled-components": "^5.3.3",
```
Comment (tupe12334) [inline packages/amplication-design-system/package.json:71]: Why is this not anchor? and why styled-components moved

File: packages/amplication-design-system/.storybook/main.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
+const path = require('path');
```
Comment (tupe12334) [inline packages/amplication-design-system/.storybook/main.js:1]: Why do we need this? its not been used

File: packages/amplication-design-system/.npmrc [hunk 6]
```diff
@@ -0,0 +1 @@
+legacy-peer-deps=true
```
Comment (tupe12334) [inline packages/amplication-design-system/.npmrc:1]: Why do we need this? its can make a breaking changes

Comment (EugeneTseitlin) [reply packages/amplication-design-system/.npmrc:1]: That was added by storybook init, I think we should keep it.

File: packages/amplication-design-system/package.json (hunk 2 above)
Comment (EugeneTseitlin) [reply packages/amplication-design-system/package.json:51]: Locked versions don't let to update packages that dependent packages. Basically that what didn't let storybook to compile and run.

File: packages/amplication-design-system/package.json (hunk 1 above)
Comment (EugeneTseitlin) [reply packages/amplication-design-system/package.json:61]: We don't have cyclic dependencies here, we don't need peer dependencies.

File: packages/amplication-design-system/package.json (hunk 3 above)
Comment (EugeneTseitlin) [reply packages/amplication-design-system/package.json:73]: Same as above.

File: packages/amplication-design-system/package.json (hunk 4 above)
Comment (EugeneTseitlin) [reply packages/amplication-design-system/package.json:71]: Same as above.

File: packages/amplication-design-system/.npmrc (hunk 6 above)
Comment (tupe12334) [reply packages/amplication-design-system/.npmrc:1]: Got that

**Comment by yuval-hazaz** [comment]
@EugeneTseitlin please fix the CI so we can merge it

**Comment by yuval-hazaz** [comment]
@EugeneTseitlin please fix the tests on this PR so we can merge it

**Comment by EugeneTseitlin** [comment]
@yuval-hazaz Tried to fix it yesterday, it didn't go well. I have to clarify the way we build design system and how we update dependencies there.

**Comment by overbit** [comment]
Now that we have nx migration almost completed, it would be good to recreate this PR leveraging the nx plugin
https://nx.dev/storybook/overview-react

**Comment by yuval-hazaz** [comment]
@overbit let's do it. 
should we close this PR?

**Comment by overbit** [comment]
@EugeneTseitlin  Shall we close this and open a new Issue to address the storybook as a new task? @yuval-hazaz

**Comment by yuval-hazaz** [comment]
> @EugeneTseitlin Shall we close this and open a new Issue to address the storybook as a new task? @yuval-hazaz

@overbit I closed it. Please open a new issue and add it to the backlog

**Pull request closed without merging**